docs/plugin-development.rst

```diff
@@ -389,6 +389,18 @@ You can use all of the testing utilities provided by the ``base.TestCase`` class
 from core. You will also get coverage results for your plugin aggregated with
 the main Girder coverage results if coverage is enabled.
 
+Plugins can also use the external data interface provided by Girder as described
+in :ref:`use_external_data`.  For plugins, the data key files should be placed
+inside a directory called ``plugin_tests/data/``.  When referencing the
+files, they must be prefixed by your plugin name as follows
+
+.. code-block:: cmake
+
+    add_python_test(my_test EXTERNAL_DATA plugins/cats/test_file.txt)
```

Comment:
So to partially answer my own question, I suppose plugins may want to reference data key files from core. However, you could just have plugin data key file loader:
1. if this is a ``PLUGIN``, first search for the specified data key file in the ``plugins/<name>/plugin_tests/data/`` directory.
2. If not found, or not a ``PLUGIN``, search for the specified data key file in the ``tests/data/`` directory.